apps/coscrad-frontend/term-index.page.tsx

```diff
@@ -0,0 +1,169 @@
+import { useAppDispatch } from './src/app/hooks';
+import {
+    ALL_PROPERTIES_SEARCH_KEY,
+    IndexSearchScope,
+} from './src/components/resources/terms/term-index-table';
+import { TermSearchBar } from './src/components/resources/terms/term-search-bar';
+import {
+    fetchTerms,
+    filterTerms,
+    IUserDefinedFilter,
+    IUserQueryOptions,
+} from './src/store/slices/resources';
+import { TermListContainer } from './term-list.container';
+
+const compileMultilingualTextContainsQuery = (
+    fieldName: keyof ITermViewModel,
+    queryString: string
+): IUserDefinedFilter<ITermViewModel> => {
+    const extractedLanguageCode = queryString.slice(1).split('}')[0];
+
+    const searchTermsWithLanguageCodeRemoved = queryString.split('}')[1];
+
+    if (Object.values(LanguageCode).some((lc) => lc === extractedLanguageCode)) {
+        return {
+            type: 'SIMPLE',
+            operator: 'MULTILINGUAL_TEXT_INCLUDES',
+            field: fieldName,
+            params: [searchTermsWithLanguageCodeRemoved, extractedLanguageCode],
+        };
+    }
+
+    /**
+     * If the language code is not a known language code, we naively search
+     * for the text, e.g., including `{foo}` in `{foo}Ooops`.
+     */
+    return {
+        type: 'SIMPLE',
+        field: fieldName,
+        operator: 'MULTILINGUAL_TEXT_INCLUDES',
+        params: [queryString],
+    };
+};
+
+const compileCoscradQueryFromUserSearchText = (
+    scope: IndexSearchScope<ITermViewModel>,
+    queryString: string
+): IUserQueryOptions<ITermViewModel>['filter'] => {
+    if (scope === ALL_PROPERTIES_SEARCH_KEY) {
+        return {
+            type: 'OR',
+            // @ts-expect-error TODO let's sort out the full types in api-interfaces
+            conditions: (['name', 'contributions', 'vocabularyLists', 'tokens'] as const).map(
+                (field) => compileCoscradQueryFromUserSearchText(field, queryString)
+            ),
+        };
+    }
+
+    if (scope === 'name') {
+        if (queryString.charAt(0) === '{' && queryString.includes('}')) {
+            return compileMultilingualTextContainsQuery(scope, queryString);
+        }
+    }
+
+    if (scope === 'contributions') {
+        const simpleFilter: IUserQueryOptions<ITermViewModel>['filter'] = {
+            type: 'SIMPLE',
+            field: `contributions[*].statement`,
+            operator: 'TEXT_INCLUDES',
+            params: [queryString],
+        };
+
+        return simpleFilter;
+    }
+
+    if (scope === 'vocabularyLists') {
+        return {
+            type: 'SIMPLE',
+            field: `vocabularyLists[*].name`,
+            operator: 'MULTILINGUAL_TEXT_INCLUDES',
+            // TODO Include language code option
+            params: [queryString],
+        };
+    }
+
+    if (scope === 'tokens') {
+        return {
+            type: 'SIMPLE',
+            field: `tokens[*].characters[*].text`,
+            // TODO support this
+            operator: 'TEXT_EQUALS',
```

Comment:
This is the operator that gave me the message: `Unsupported logical operator for COSCRAD query filter: TEXT_EQUALS`